go/vt/vttablet/tabletmanager/rpc_vreplication.go

```diff
@@ -62,6 +62,26 @@ const (
 	sqlUpdateVReplicationWorkflows = "update /*vt+ ALLOW_UNSAFE_VREPLICATION_WRITE */ %s.vreplication set%s where db_name = '%s'%s"
 	// Check if workflow is still copying.
 	sqlGetVReplicationCopyStatus = "select distinct vrepl_id from %s.copy_state where vrepl_id = %d"
+	// Validate the minimum set of permissions needed to manage vreplication metadata.
+	// This is a simple check for a matching user rather than any specific user@host
+	// combination.
+	sqlValidateVReplicationPermissions = `
+select if(count(*)>0, 1, 0) as good from mysql.user as u
+  left join mysql.db as d on (u.user = d.user)
+  left join mysql.tables_priv as t on (u.user = t.user)
+where u.user = %a
+  and (
+    (u.select_priv = 'y' and u.insert_priv = 'y' and u.update_priv = 'y' and u.delete_priv = 'y') /* user has global privs */
+    or (d.db = %a and d.select_priv = 'y' and d.insert_priv = 'y' and d.update_priv = 'y' and d.delete_priv = 'y') /* user has db privs */
+    or (t.db = %a and t.table_name = 'vreplication' /* user has table privs */
+      and find_in_set('select', t.table_priv)
+      and find_in_set('insert', t.table_priv)
+      and find_in_set('update', t.table_priv)
+      and find_in_set('delete', t.table_priv)
+    )
+  )
+limit 1
+`
```

Comment:
This looks correct. I wonder if we should not instead parse the result of `SHOW GRANTS` instead, as those guarantee correct aggregation of the above `mysql.*` tables.

Reply:
I thought about that as well, but I thought that it would be much more complicated since I didn't see any existing parser support for it. And since that output is formulated from querying these tables anyway I went this way.